packages/file_selector/file_selector_android/android/src/main/java/io/flutter/plugins/file_selector/FileSelectorPlugin.java

```diff
@@ -0,0 +1,126 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.file_selector;
+
+import android.app.Activity;
+import android.app.Application;
+import android.util.Log;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
+import io.flutter.embedding.engine.plugins.FlutterPlugin;
+import io.flutter.embedding.engine.plugins.activity.ActivityAware;
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding;
+import io.flutter.plugin.common.BinaryMessenger;
+import io.flutter.plugins.file_selector.Messages.FileSelectorApi;
+import java.util.List;
+
+/** Android platform implementation of the FileSelectorPlugin. */
+public class FileSelectorPlugin implements FlutterPlugin, FileSelectorApi, ActivityAware {
+
+  static final String TAG = "FileSelectorPlugin";
+
+  FlutterPluginBinding pluginBinding;
+  ActivityStateHelper activityState;
+  FileSelectorDelegate delegate;
+
+  /**
+   * Default constructor for the plugin.
+   *
+   * <p>Use this constructor for production code.
+   */
+  public FileSelectorPlugin() {}
+
+  @VisibleForTesting
+  FileSelectorPlugin(final FileSelectorDelegate delegate, final Activity activity) {
+    activityState = new ActivityStateHelper(delegate, activity);
+  }
+
+  final ActivityStateHelper getActivityState() {
+    return activityState;
+  }
+
+  private void setup(
```

Comment:
`setUp`; setup is a noun, not a verb, and methods should generally have verb names. (The Pigeon naming is a bug in Pigeon.)